api/filters/replacement/replacement.go

```diff
@@ -178,3 +178,27 @@ func makeResId(n *yaml.RNode) *resid.ResId {
 		Namespace: n.GetNamespace(),
 	}
 }
+
+func getFieldPath(fieldPath string) []string {
```

Comment:
Please 

 - move this function to https://github.com/kubernetes-sigs/kustomize/blob/master/api/internal/utils/pathsplitter.go alongside the func called PathSplitter.
 - call it `SmarterPathSplitter` to distinguish it from `PathSplitter`.
 - optionally pass the delimitter, so we can pass `'.'` or `'/'` (pass a byte, not a string, so we can someday respect escaping of the delimiter)
 

Comment could be:

```
// SmarterPathSplitter splits a path, retaining bracketed list entry identifiers.
// E.g. [name=com.foo.someapp] survives as one thing after splitting
// "spec.template.spec.containers.[name=com.foo.someapp].image"
// See kyaml/yaml/match.go for use of list entry identifiers.
// TODO combine with `PathSplitter`, so one function both respects list entry identifiers
// and escaped delimiters
```

Reply:
I just saw @KnVerey 's comment.

before moving to kyaml, please first combine it with the existing PathSplitter somehow (respected escaping) and then move it.   perhaps in a distinctr Pr after this.

Another possible home for this splitter (these functions) is as a method on the FieldSpec object.

Reply:
BTW, `SmarterPathSplitter` is silly, but `ListEntryIdentifierAwarePathSplitter` is a bit long.

I'd prefer just to have one pathsplitter to avoid making up a new name :)

Reply:
I moved it to api/internal/utils/pathsplitter.go and also had it use the existing `PathSplitter` so that SmarterPathSplitter respects both bracketed list elements as well as escaped delimiters. I'll move it to kyaml in a follow-up.